### PR TITLE
test/document custom get/set form elements

### PR DESCRIPTION
**Pull request by MykolaGolubyev**

### CODE REVIEW

**Comment by MykolaGolubyev** [comment]
@tsiq-karold please review docs

File: webtau-browser/src/main/java/com/twosigma/webtau/browser/page/path/GenericPageElement.java [hunk 1]
```diff
@@ -171,6 +171,11 @@ public String getText() {
         return findElement().getText();
     }
 
+    public String getUnderlyingValue() {
```
Comment (tsiq-karold) [inline webtau-browser/src/main/java/com/twosigma/webtau/browser/page/path/GenericPageElement.java]: `@Override`?

File: webtau-browser/src/main/java/com/twosigma/webtau/browser/page/path/GenericPageElement.java [hunk 2]
```diff
@@ -207,7 +207,7 @@ private String getUnderlyingValue() {
 
         for (int idx = 0; idx < elements.size(); idx++) {
             HtmlNode htmlNode = new HtmlNode(elementsMeta.get(idx));
-            PageElement pageElementByIdx = get(idx);
+            PageElement pageElementByIdx = get(idx + 1);
```
Comment (tsiq-karold) [inline webtau-browser/src/main/java/com/twosigma/webtau/browser/page/path/GenericPageElement.java]: Love off by one errors 😄 Is there a test that exercises this?

File: webtau-docs/webtau/UI/forms.md [hunk 3]
```diff
@@ -33,3 +33,28 @@ Form element value can be asserted the same way as any regular element.
...
 :include-file: doc-artifacts/snippets/forms/validation.groovy {title: "universal assert"}
+
+# Custom Form Elements
+
+One of the benefits of universal `set` and `assert` is that your test is focused on the data and not implementation details.
+But what if you decided to use a custom component to enter the data? 
+
+To hide implementation details from your test you should define a custom input handler for your UI component.
+
+Let's consider a form component that you can start interacting with only after you clicked it. And after the value
```
Comment (tsiq-karold) [inline webtau-docs/webtau/UI/forms.md]: `the input box`

File: webtau-browser/src/main/java/com/twosigma/webtau/browser/page/path/GenericPageElement.java (hunk 2 above)
Comment (tsiq-nick) [reply webtau-browser/src/main/java/com/twosigma/webtau/browser/page/path/GenericPageElement.java]: there is end to end one that was broken.

**Pull request merged**